modules/@apostrophecms/area/ui/apos/components/AposWidgetControls.vue

```diff
@@ -124,54 +106,137 @@ export default {
+        disabled: this.last || this.disabled,
+        tooltip: {
+          content: this.last || this.disabled ? null : 'apostrophe:nudgeDown',
+          placement: 'left'
+        },
+        action: 'down'
+      });
+
+      // Edit
+      if (!this.options.contextual) {
+        controls.push({
+          ...this.widgetDefaultControl,
+          label: 'apostrophe:edit',
+          icon: 'playlist-edit-icon',
+          disabled: this.disabled,
+          tooltip: {
+            content: 'apostrophe:editWidget',
+            placement: 'left'
+          },
+          action: 'edit'
+        });
+      }
+
+      // Custom widget operations displayed in the primary controls
+      controls.push(
+        ...this.widgetPrimaryOperations.map(operation => ({
+          ...this.widgetDefaultControl,
+          ...operation,
+          disabled: this.disabled,
+          tooltip: {
+            content: operation.label,
+            placement: 'left'
+          }
+        }))
+      );
+
+      return controls;
     },
-    cloneButton() {
-      return {
-        ...this.buttonDefaults,
-        label: 'apostrophe:clone',
-        icon: 'content-copy-icon'
-      };
+    widgetSecondaryControls() {
+      const controls = [];
+
+      // Cut
+      controls.push({
+        label: 'apostrophe:cut',
+        icon: 'content-cut-icon',
+        action: 'cut'
+      });
+
+      // Copy
+      controls.push({
+        label: 'apostrophe:copy',
+        icon: 'content-copy-icon',
+        action: 'copy'
+      });
+
+      // Clone
+      controls.push({
+        label: 'apostrophe:duplicate',
+        icon: 'content-duplicate-icon',
+        action: 'clone',
+        modifiers: [
+          ...(this.disabled || this.maxReached) ? [ 'disabled' ] : []
+        ]
+      });
+
+      if (this.widgetSecondaryOperations.length) {
+        controls.push({
+          separator: true
+        });
+      }
+
+      // Custom widget operations displayed in the secondary controls
+      controls.push(...this.widgetSecondaryOperations);
+
+      return controls;
     },
-    removeButton() {
+    widgetRemoveControl() {
       return {
-        ...this.buttonDefaults,
+        ...this.widgetDefaultControl,
         label: 'apostrophe:remove',
-        icon: 'trash-can-outline-icon'
-      };
-    },
-    editButton() {
-      return {
-        ...this.buttonDefaults,
-        label: 'apostrophe:edit',
-        icon: 'pencil-icon'
-      };
-    },
-    cutButton() {
-      return {
-        ...this.buttonDefaults,
-        label: 'apostrophe:cut',
-        icon: 'content-cut-icon'
+        icon: 'trash-can-outline-icon',
+        disabled: this.disabled,
+        tooltip: {
+          content: 'apostrophe:delete',
+          placement: 'left'
+        },
+        action: 'remove'
       };
+    }
+  },
+  methods: {
+    filterByWidgetType(operation) {
+      return !operation.type || operation.type === this.modelValue.type;
     },
```

Comment:
This method is never used, you could use it in data

Reply:
It was used.
But now it's been moved to the shared method, see 5873267c9cb2395cc13aed36e3479cb833615bdf

Reply:
Erratum.
It wasn't used 👍